internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -44,6 +44,11 @@ var (
 func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) (ctrl.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
 
+	// Create a watch on the nodes in the Cluster.
+	if err := r.watchClusterNodes(ctx, cluster); err != nil {
```

Comment:
@fabriziopandini I think we should be alright to add the tracker where it's needed instead of ignoring the error earlier in the reconcile.

Alternatively we could ignore the error but check in this function if the watch is actually set and exit early if not.

Reply:
I don't have strong objections to this specific change

However, as a follow up, I think we should probably think more holistically about how failures on getting a client from the tracker are handled, because now all the reconciler trying to get an accessor currently locked goes into exponential backoff, and this has two downsides:
- we have a spike of errors on the logs
- exponential backoff grows fast, quickly going over the resync period. 

I can see two options here: 
- a quick win, which is requeing after the timeout for creating a new accessor
- a better solution, where the cluster tracker enquees events for the reconcilers when a new accessor is created.

@sbueringer opinions?

Reply:
The change in this PR looks good to me

Reply:
The default max exponential backoff is ~16 minutes  45 seconds for controller runtime, and we currently use that default across our controllers. I'm going to and open an issue that points to this thread so we can discuss this in a more visible place. 

I do think we should be concerned with how often we return errors/requeues and resort to exponential backoff when it comes to total time for operations to complete vs stability. There may be cases where we want to err on the side of less or more frequent reconciles depending on the operation.